query/network/network_security_group_not_configured_gateway_subnets.sql

```diff
@@ -0,0 +1,19 @@
+select
+  -- Required Columns
+  subnet.id resource,
+  case
+    when nat_gateway_id is null then 'skip'
```

Comment:
we have to check if the the subnet with name equals to `GatewaySubnet` is assigned with `networkSecurityGroup id ` then is should throw error, please check once.
```       {
            "field": "type",
            "equals": "Microsoft.Network/virtualNetworks/subnets"
          },
          {
            "field": "name",
            "equals": "GatewaySubnet"
          },
          {
            "field": "Microsoft.Network/virtualNetworks/subnets/networkSecurityGroup.id",
            "exists": "true"
          }
          ```